### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by alex31425**

### CODE REVIEW

File: ML.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from sklearn.datasets import load_iris
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.model_selection import train_test_split
+from sklearn import metrics
+from sklearn.pipeline import Pipeline
+import pandas as pd
+import numpy as np
+
+# data source : https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_iris.html
```
Comment (michelle-an) [inline ML.py:11]: I like the description to put the source of data.

File: ML.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from sklearn.preprocessing import Normalizer
+from sklearn.datasets import load_iris
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.model_selection import train_test_split
+from sklearn import metrics
+from sklearn.pipeline import Pipeline
+import pandas as pd
+import numpy as np
```
Comment (michelle-an) [inline ML.py:9]: It's interesting that you split the statistical analysis and machine learning separately! I like it

File: ML.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+X = normalized
+
+y = iris.target
+
+X_train, X_test, y_train, y_test = train_test_split(X, y)
+clf=RandomForestClassifier()
+clf.fit(X_train,y_train)
+y_pred=clf.predict(X_test)
+print("Accuracy:",metrics.accuracy_score(y_test, y_pred))
+
```
Comment (michelle-an) [inline ML.py:30]: This looks good, but wrap it in a pipeline for reproducibility and call it using pipeline.fit(X,y). 
Also, do one more ML classifier (ie. SVM, KNN, etc). 
Once one ML classifier is working, it's easy to copy the code and do the same for the 2nd.

File: ML.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+X = normalized
+
+y = iris.target
+
+X_train, X_test, y_train, y_test = train_test_split(X, y)
+clf=RandomForestClassifier()
+clf.fit(X_train,y_train)
+y_pred=clf.predict(X_test)
+print("Accuracy:",metrics.accuracy_score(y_test, y_pred))
```
Comment (michelle-an) [inline ML.py:29]: I like that you do the accuracy score!

File: main.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+\nQ1 quantile of setosa is {np.quantile(np.array(setosa), 0.25)}\
+\nQ3 quantile of setosa is {np.quantile(np.array(setosa), 0.75)}')
+print('versicolor')
+print(f'Q2 quantile of versicolor is {np.quantile(np.array(versicolor), 0.5)}\
+\nQ1 quantile of versicolor is {np.quantile(np.array(versicolor), 0.25)}\
+\nQ3 quantile of versicolor is {np.quantile(np.array(versicolor), 0.75)}')
+print('virginica')
+print(f'Q2 quantile of virginica is {np.quantile(np.array(virginica), 0.5)}\
+\nQ1 quantile of virginica is {np.quantile(np.array(virginica), 0.25)}\
+\nQ3 quantile of virginica is {np.quantile(np.array(virginica), 0.75)}')
```
Comment (michelle-an) [inline main.py:71]: I like the use of f strings to see the clear outputs of the analysis.

File: main.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+for i in range(len(arr)):
+    if arr[i][4] == U_name[0]:
+        for n in arr[i][0:4]:
+            setosa.append(n)
+    elif arr[i][4] == U_name[1]:
+        for n in arr[i][0:4]:
+            versicolor.append(n)
+    else :
+        for n in arr[i][0:4]:
+            virginica.append(n)
```
Comment (michelle-an) [inline main.py:43]: No need to clean this up, but if you want to split the dataset by species types, I think there are quicker ways to do this by chopping up an array into smaller sub arrays based on the value of a single column (ie. creating a temporary boolean array).

**Review by michelle-an — CHANGES_REQUESTED** [review]
Looks good, but I think for the assignment we need to assess two types of ML classifiers. Also I advise to wrap them in pipelines. At the end, make sure to use linters for better formats and suggestions! 👍

File: main.py (hunk 6 above)
Comment (alex31425) [reply main.py:43]: Good idea! I am getting familiar with Numpy

**Review by michelle-an — APPROVED** [review]
Looks good!
